speechbrain/utils/distributed.py

```diff
@@ -126,11 +126,12 @@ def ddp_init_group(run_opts):
                 "--distributed_backend=nccl"
             )
         else:
-            if run_opts["local_rank"] + 1 > torch.cuda.device_count():
-                raise ValueError(
-                    "Killing process " + str() + "\n"
-                    "Not enough GPUs available!"
-                )
+            if not run_opts["distributed_backend"] == "gloo":
```

Comment:
Again, why is gloo treated like that? :p

Reply:
the way it is implemented for GPU is fully incompatible with GPU, in fact: gloo will crash when run in cpu only.

This pytorch example for gloo is incompatible with SB, as it was before the contributed fix
https://pytorch.org/tutorials/intermediate/dist_tuto.html

Reply:
for this part, if there is only CPU, ofc there are not enough GPUs ;)